### PR TITLE
Credorax: add rich 3DS support

### PR DESCRIPTION
**Pull request by bpollack**

### CODE REVIEW

**Comment by curiousepic** [comment]
Let's add the new fields to scrub.

**Comment by curiousepic** [comment]
Well, I've never been clear on whether they really should be, curious to hear your opinion on that.

**Comment by bpollack** [comment]
Well, at a bare minimum, I'm passing the CC in the MPI requests, which definitely needs to be scrubbed, so…let's start there. :)

**Comment by curiousepic** [comment]
I think the unit tests look good, though with the external MPI craziness I can't quite tell if there's a permutation of request/response scenarios that isn't tested. :not_sure_if:

**Comment by bpollack** [comment]
@curiousepic Scrubbing added, plus some cleanup to reflect how `three_d_options` is usually passed.

**Comment by bpollack** [comment]
(And I also added an additional remote tests, but I think this is as far as it goes.)

File: test/remote/gateways/remote_credorax_test.rb [hunk 1]
```diff
@@ -168,6 +186,17 @@ def test_transcript_scrubbing
     assert_scrubbed(@credit_card.verification_value.to_s, clean_transcript)
   end
 
+  def test_3ds_transcript_scrubbing
+    transcript = capture_transcript(@gateway) do
+      @gateway.purchase(@amount, @credit_card, @options.merge({three_d_secure: {}, xid: '12345678901234567890'}))
+    end
+    clean_transcript = @gateway.scrub(transcript)
+    puts "!!!!!!\n#{clean_transcript}\n!!!!!"
```
Comment (curiousepic) [inline test/remote/gateways/remote_credorax_test.rb]: Do we want this here?

**Comment by bpollack** [comment]
> I think the unit tests look good, though with the extra MPI craziness I can't quite tell if there's a permutation of request/response scenarios that isn't covered (that could be). :not_sure_if:

It might be a good exercise for both of us if you could articulate anything obviously missing, even vaguely; I don't exactly have super-high confidence in this code.

File: test/remote/gateways/remote_credorax_test.rb (hunk 1 above)
Comment (bpollack) [reply test/remote/gateways/remote_credorax_test.rb]: 🤦‍♂️ No. Removing.

**Comment by bpollack** [comment]
@curiousepic Nuked the erroneous `puts`, plus fixed the RuboCop error.

**Comment by bpollack** [comment]
@nfarve Yeah. I'm going to retract this; it's been subsumed by the Core PR entirely.

**Pull request closed without merging**